R/utils.R

```diff
@@ -20,3 +20,24 @@ log_into_synapse <- function() {
   syn <- synapse$login()
   syn
 }
+
+#' Check that a value is a positive integer
+#'
+#' Check that a value is a positive integer. If the value is a string, will
+#' return `TRUE` if it can be coerced to a positive integer.
+#'
+#' @param data A single value to check whether it is a positive integer.
+#' @return `TRUE`` if data is an integer or a string that is coercible to a
+#' positive integer; else `FALSE`
+is_positive_integer <- function(data) {
+  numeric_data <- suppressWarnings(as.numeric(data))
+  if (is.na(data) || is.null(data) || is.na(numeric_data)) {
+    return(FALSE)
+  }
+  if (inherits(data, "numeric") && data > 0 && data %% 1 == 0 ||
+    inherits(data, "character") && numeric_data > 0 && numeric_data %% 1 == 0) {
```

Comment:
Since we've already created a version of the data that is sure to be numeric, could this if statement use that rather than having separate options for numeric `data` vs. character `data`?

Reply:
Need to still check for integers since numeric doesn't cover integers. Can remove data checks since already checking that numeric_data is positive integer. Keep checks for numeric and character because, for example, booleans can be converted to numeric.